src/oracle/oracleDataService.ts

```diff
@@ -0,0 +1,82 @@
+import pool from "../common/pool";
+import logFn from "../common/logger";
+import { OracleStatus } from "./oracleModel";
+import { ApiPromise } from "@polkadot/api";
+import { hexStringFixedPointToBig } from "../common/util";
+
+export const logger = logFn({ name: "oracleDataService" });
+
+function computeOfflineStatusThreshold(maxDelay: number): Date {
+    return new Date(Date.now() - maxDelay);
+
+}
+
+export async function getLatestSubmissionForEachOracle(
+    api: ApiPromise,
+    maxDelay: number,
+    feed: string,
+    namesMap: Map<string, string>
+): Promise<OracleStatus[]> {
+    try {
+        const res = await pool.query(`
+            SELECT DISTINCT oracle_id, exchange_rate, block_ts
+            FROM v_parachain_oracle_set_exchange_rate main
+            WHERE block_ts = (
+                SELECT MAX(block_ts) 
+                FROM v_parachain_oracle_set_exchange_rate v 
+                WHERE v.oracle_id = main.oracle_id
+            )
+        `);
+        const offlineStatusThreshold = computeOfflineStatusThreshold(maxDelay);
+        return Promise.all(res.rows.map(async (row) => {
+            const submissionMilliseconds = new Date(row.block_ts).getTime();
+            const offlineOracleMilliseconds = offlineStatusThreshold.getTime();
+            const online = submissionMilliseconds >= offlineOracleMilliseconds;
+            return {
+                id: row.oracle_id,
+                source: namesMap.get(row.oracle_id) || "",
+                feed,
+                lastUpdate: new Date(row.block_ts),
+                exchangeRate: hexStringFixedPointToBig(api, row.exchange_rate).toString(),
```

Comment:
I guess this call would not need to call the parachain technically so not sure exactly why the API needs to be part of this signature. @theodorebugnet is there a specific reason for this?

Reply:
The API is used to construct a `FixedI128`. Possibly this could be avoided, with or without modifying some lib function signatures, but it was the most straightforward way to achieve a conversion and has no significant cost that I'm aware of

Reply:
I think it might explain though why some stats calls are slow since it needs to establish the was connection to the chain first instead of "just" having a SQL query.

Reply:
Wait, does `api.createType` actually connect to the parachain?

Reply:
It can be removed if we instantiate with an empty `TypeRegistry` object. I updated

Reply:
I think we should have a look at the other calls as well. Because if we do a new was connection on some of these DB calls, that will slow the response down